src/librustdoc/visit_ast.rs

```diff
@@ -337,4 +352,58 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
             om.foreigns.push((item, renamed));
         }
     }
+
+    /// Run a few checks on the `#[doc]` attributes for an item:
+    /// - warn if a crate-level attribute is used on a non-`crate` item
+    /// - warn if `#[doc(inline)]`/`#[doc(no_inline)]` is used on a non-`use` item
+    /// - error if conflicting `#[doc(inline)]`/`#[doc(no_inline)]` are found
+    fn check_item_attrs(&mut self, attrs: &'tcx [ast::Attribute], is_crate: bool, is_use: bool) {
+        let mut specified_inline = None;
+
+        for attr in attrs.lists(sym::doc) {
+            // Check for crate-level attributes used on non-crate item
+            if !is_crate
+                && attr
+                    .meta_item()
+                    .map_or(false, |meta| CRATE_LEVEL_ATTRS.iter().any(|attr| meta.path == *attr))
+            {
+                self.cx
+                    .sess()
+                    .struct_span_warn(attr.meta_item().unwrap().span, "this attribute can only be applied at the crate level")
+                    .note("read https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#at-the-crate-level for more information")
+                    .emit();
+            }
+
+            // Check inline attributes
+            if attr.has_name(sym::inline) || attr.has_name(sym::no_inline) {
+                if is_use {
+                    let inline = attr.has_name(sym::inline);
+                    if let Some((prev_inline, prev_span)) = specified_inline {
+                        if prev_inline != inline {
+                            let attr_span = attr.meta_item().unwrap().span;
+                            let mut spans = MultiSpan::from_spans(vec![prev_span, attr_span]);
+                            spans.push_span_label(prev_span, String::from("this attribute..."));
+                            spans.push_span_label(
+                                attr_span,
+                                String::from("...conflicts with this attribute"),
+                            );
+                            self.cx
+                                .sess()
+                                .struct_span_err(spans, "conflicting doc inlining attributes")
```

Comment:
What's the difference between using `MultiSpan` and `diag.span_label("this attribute")`?

Reply:
Also, I'm think this isn't supposed to depend on the order of the attributes in case the diagnostic library switches it.

Reply:
IIUC, `span_label` adds a "non-primary" span which getd shown in blue. Here boths spans get shown in red.
Unsure what's best though.

Reply:
Ahh, that makes sense - you have two primary labels instead of one primary and one secondary. No need to change this, then.